project/src/main/scala/locales/ScalaLocaleCodeGen.scala

```diff
@@ -169,6 +185,18 @@ object CodeGenerator {
         percent, minus, perMille, infinity, nan, exp)
     }
 
+    val currencies = ldml.currencies.map{ c =>
+      val symbols = LIST(c.symbols.map{ s =>
+        Apply(ldmlCurrencySym, LIT(s.symbol), LITOPTION(s.alt))
+      })
+
+      val displayNames = LIST(c.displayNames.map{ n =>
+        Apply(ldmlCurrencyDisplayName, LIT(n.name), LITOPTION(n.count))
+      })
+
+      Apply(ldmlNumberCurrency, LIT(c.currencyCode), symbols, displayNames)
+    }
+
```

Comment:
You may need to if you want to support methods like `NumberFormat.getCurrency`

https://docs.oracle.com/javase/7/docs/api/java/text/NumberFormat.html#getCurrency()